src/hardware/tandy_sound.cpp

```diff
@@ -631,26 +630,66 @@ static void set_tandy_sound_flag_in_bios(const bool is_enabled)
 	real_writeb(0x40, 0xd4, is_enabled ? 0xff : 0x00);
 }
 
-static void shutdown_dac(Section*)
+static ConfigProfile get_config_profile()
 {
+	switch (machine) {
+	case MCH_PCJR: return ConfigProfile::PCjrSystem; break;
+	case MCH_TANDY: return ConfigProfile::TandySystem; break;
+	default: return ConfigProfile::SoundCardOnly; break;
+	}
+}
+
+static void toggle_dac(Section* section)
+{
+	static std::unique_ptr<TandyDAC> tandy_dac_instance = {};
+
+	// Always reset the Tandy DAC before configuring
 	if (tandy_dac) {
 		LOG_MSG("TANDY: Shutting down DAC");
-		tandy_dac.reset();
+		tandy_dac = nullptr;
+		tandy_dac_instance.reset();
+	}
+	// If the section is empty, the DAC isn't wanted
+	if (!section) {
+		return;
+	}
+
+	assert(section);
+	const auto prop = static_cast<Section_prop*>(section);
+	const auto pref = std::string_view(prop->Get_string("tandy"));
+
+	const auto wants_dac = has_true(pref) || (IS_TANDY_ARCH && pref == "auto");
+	if (wants_dac) {
+		const auto config_profile = get_config_profile();
+
+		tandy_dac_instance = std::make_unique<TandyDAC>(
+		        config_profile, prop->Get_string("tandy_dac_filter"));
+
+		tandy_dac = tandy_dac_instance.get();
 	}
 }
 
-void TANDYSOUND_ShutDown(Section*)
+void TANDYSOUND_Init(Section* section)
 {
-	if (tandy_psg || tandy_dac) {
-		LOG_MSG("TANDY: Shutting down");
+	// The Tandy DAC and PSG (programmable sound generator) managed pointers
+	static std::unique_ptr<TandyPSG> tandy_psg_instance = {};
+
+	// Always reset the Tandy PSG before configuring
+	if (tandy_psg_instance) {
+		LOG_MSG("TANDY: Shutting down PSG");
 		set_tandy_sound_flag_in_bios(false);
-		tandy_dac.reset();
-		tandy_psg.reset();
+		tandy_psg_instance.reset();
+	}
+
+	// Also reset the Tandy DAC with an empty section
+	constexpr Section* empty_section = nullptr;
+	toggle_dac(empty_section);
+
+	// If the section's empty, then there's nothing to configure
+	if (!section) {
```

Comment:
I'm a bit confuse why this is needed. To guard against programming errors? Unless I'm wrong, `section` can only be null if someone removed the whole section from the config, so this should rather be an assert on top of the function, no?